src/server/game/Entities/Object/Object.cpp

```diff
@@ -1147,7 +1147,12 @@ bool WorldObject::IsWithinLOSInMap(const WorldObject* obj) const
         return false;
 
     float ox, oy, oz;
-    obj->GetPosition(ox, oy, oz);
+    float angle = obj->GetAngle(GetPositionX(), GetPositionY());
+    float range = std::min(GetObjectSize(), GetExactDist(obj->GetPositionX(), obj->GetPositionY(), obj->GetPositionZ()));
```

Comment:
It taking the bottom position of the objects and checks if those are in los (which, in the case of the screenshot, are not.. :/)